app/controllers/api/v2/search_controller.rb

```diff
@@ -18,12 +19,22 @@ def index
 
   private
 
+  def validate_search_params!
+    params.require(:q)
+
+    return if user_signed_in?
+
+    render json: { error: 'Search queries for fewer than 5 characters are not supported without authentication' }, status: 401 if params[:q].size < 5
```

Comment:
We might want to allow exact matches, otherwise, say `@bob` wouldn't be searchable.